### ISSUE TITLE
Fix misplaced items in 'Breaking changes' page

### ISSUE THREAD

**Issue by rydmike** [flutter/website#8585]
### Page URL

https://docs.flutter.dev/release/breaking-changes

### Describe the problem

# Page: Breaking changes 

The [Breaking changes](https://docs.flutter.dev/release/breaking-changes) page lists or misses the following `ThemeData` color deprecations:

## Version 3.7 - Wrong version on ThemeData color property deprecation info

* `toggleableActiveColor` is listed as deprecated in version 3.7, it was in fact deprecated in version 3.3 already. 
See https://github.com/flutter/flutter/blob/3.3/packages/flutter/lib/src/material/theme_data.dart#L421

   Which states *'This feature was deprecated after v2.13.0-0.4.pre’*. Later in current stable the statement was changed to say *'This feature was deprecated after v3.4.0-19.0.pre’*. If that was the case it would not have landed in Flutter 3.3, but it was in fact deprecated and already included in **Flutter stable 3.3**. The deprecation annotation and notice was there, and we had to deal with already in version 3.3.

## Version 3.7 - Missing ThemeData color property deprecation info

Current stable Flutter 3.7 also deprecated these `ThemeData` color properties, but the info is missing.

* `errorColor`: See https://github.com/flutter/flutter/blob/stable/packages/flutter/lib/src/material/theme_data.dart#L444

* `backgroundColor`: See https://github.com/flutter/flutter/blob/stable/packages/flutter/lib/src/material/theme_data.dart#L449

* `bottomAppBarColor`: See https://github.com/flutter/flutter/blob/stable/packages/flutter/lib/src/material/theme_data.dart#L454

## Version 3.3 - Missing ThemeData color property deprecation info

* `toggleableActiveColor` as mentioned above, it should be included in this version already.

* `selectedRowColor` was also also deprecated already in version 3.3, the info is missing. See https://github.com/flutter/flutter/blob/3.3/packages/flutter/lib/src/material/theme_data.dart#L426

### Expected fix

Expect above mentioned documented deprecations, versus actual deprecations, to be corrected on the https://docs.flutter.dev/release/breaking-changes page.

**Comment by atsansone** [comment]
@itsjustkevin : Can you confirm these changes are needed? I'll fix when I get your OK.

**Comment by itsjustkevin** [comment]
Oof, this is a hefty one @atsansone!  I will take a look tomorrow.  

@rydmike thanks as always for reporting possible issues.

**Comment by itsjustkevin** [comment]
> toggleableActiveColor is listed as deprecated in version 3.7, it was in fact deprecated in version 3.3 already.
See https://github.com/flutter/flutter/blob/3.3/packages/flutter/lib/src/material/theme_data.dart#L421

The breaking change page is correct on this one.  What you're seeing here is the 3.3 branch which is not where the actual release is.  To view where the release is at a current point, you would have to look at the tags, in this case [3.3.0](https://github.com/flutter/flutter/blob/3.3.0/packages/flutter/lib/src/material/theme_data.dart).

The release after the v3.4.0-19.0.pre branch was Flutter 3.7 which is where the deprecation became active on the stable branch.

> toggleableActiveColor as mentioned above, it should be included in this version already.

This was not available until 3.7.0.  Another way to hunt this down is to take a look at the [commit](https://github.com/flutter/flutter/commit/52829523d648733c25951a193052db79c1a54750) where the deprecation work occurred and look at the tag list and verify that the release question is in the range of tags.

This can also be accomplished by using `git tag --contains 52829523d648733c25951a193052db79c1a54750 | sort -V` which shows all tags which contain the commit in a reverse older, showing oldest to newest.  You can see here that the first stable tag this was included in is 3.7.0.

> selectedRowColor was also also deprecated already in version 3.3, the info is missing. See https://github.com/flutter/flutter/blob/3.3/packages/flutter/lib/src/material/theme_data.dart#L426

This deprecation does seem like it did not take in [3.3](https://github.com/flutter/flutter/blob/3.3.0/packages/flutter/lib/src/material/theme_data.dart), however it does appear in [3.7](https://github.com/flutter/flutter/blob/3.7.0/packages/flutter/lib/src/material/theme_data.dart#L439).

> errorColor: See https://github.com/flutter/flutter/blob/stable/packages/flutter/lib/src/material/theme_data.dart#L444

backgroundColor: See https://github.com/flutter/flutter/blob/stable/packages/flutter/lib/src/material/theme_data.dart#L449

The two above were in fact deprecated in 3.7.  @atsansone these are relevant and do need to be added if they are not on the breaking change and deprecations page. 

> bottomAppBarColor: See https://github.com/flutter/flutter/blob/stable/packages/flutter/lib/src/material/theme_data.dart#L454

I can't seem to find this deprecation.

@Piinks @guidezpl and @TahaTesser could you also help confirm these changes?

**Comment by guidezpl** [comment]
- `toggleableActiveColor` was [deprecated](https://github.com/flutter/flutter/commit/52829523d648733c25951a193052db79c1a54750) in 3.4, therefore appears in 3.7. 
- `errorColor` and `backgroundColor` were [deprecated](https://github.com/flutter/flutter/commit/8853926601d82dc350a3af8fc5e63e58f55c415b) but is not considered a breaking change
- `bottomAppColor` was indeed [deprecated](https://github.com/flutter/flutter/commit/1ed2312493ac35c17a01cf5ef8e5c9f2a0bcbf59) but is not considered a breaking change
- `selectedRowColor` was [deprecated](https://github.com/flutter/flutter/commit/8c44c162685ab1976baa897599c1441d95f09af2) in 3.4, therefore should appear in 3.7.

So to sum up, I think the only change required is documenting the `selectedRowColor` deprecation if it is a breaking change (i.e. not fully handled by Dart fix).

**Comment by guidezpl** [comment]
I thought if `dart fix` could cleanly migrate someone off old APIs (i.e. the only changes to framework tests were achieved through `dart fix`), a guide wasn't necessary, hence my loose use of the expression "not considered breaking". 

But I prefer your idea of having guides in these cases nonetheless. How about we update https://github.com/flutter/flutter/wiki/Tree-hygiene#deprecations to make the creation of a guide at deprecation time an obligatory step?

**Comment by atsansone** [comment]
@Piinks : Is the `selectedRowColor` deprecation a breaking change?

**Comment by Piinks** [comment]
Looks like it is currently deprecated and is supported by dart fix, if there isn't a migration guide, one will be added when it is removed.

**Comment by atsansone** [comment]
No migration guides are needed at this time as they either have guides or are covered by `dart fix`. Closing this issue.

**Comment by rydmike** [comment]
@atsansone and @itsjustkevin, I noticed the above listed deprecations in the Flutter version as mentioned above in original post. Because I had to deal with them already then in those versions, basically remove their usage. Because if you don't, and leave a deprecated property referenced in a package, then `pub.dev` penalizes the package score for using deprecated Flutter SDK properties.

In fact, you cannot even use package self-deprecated members without being hit by this pub score penalty. This happens even if you in you `analyzis_options.yaml` use `deprecated_member_use_from_same_package: ignore`. See https://github.com/dart-lang/sdk/issues/49258

>  Deprecations are considered breaking changes, they just haven't broken yet.

> It's important to provide these guides for deprecations to help folks migrate off the old APIs - so they can avoid being broken later when we remove deprecations. Or, if someone is upgrading after a long spell and missed the deprecation period, they will at that point just be broken.

Agreed.

> @rydmike, not all deprecations are listed under breaking changes. When they are removed from the framework though, they are always announced, included in the release notes, and have their own migration guide like: https://docs.flutter.dev/release/breaking-changes/3-7-deprecations which covers any that did not have a guide to begin with.

@Piinks, yes removal of a deprecated property, is certainly always API breaking and is always nicely reported, thank you! 💙 

Often the deprecation of a property, while maybe sometimes not breaking the code, like for above props, still in above cases requires you to do other things to retain past colors and styles. At least if you had styled your app in other ways than using SDK default colors for the above deprecated color properties. So the deprecations actually do break the style of your app in these cases, and it happened already when the deprecations were made.

Not documenting deprecations in the version as and when they are deprecated, seems a bit lazy and sloppy to me. It does not take so much to write a short notice that in this version the new deprecations are x, y, z, but sure practices varies.
